app/src/main/java/com/ddoczi/tasky/MainActivity.kt

```diff
@@ -5,33 +5,70 @@ import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.viewModels
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.collectAsState
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
 import com.ddoczi.tasky.authentication.presentation.login.LoginScreen
 import com.ddoczi.tasky.authentication.presentation.login.LoginViewModel
+import com.ddoczi.tasky.authentication.presentation.registration.RegistrationScreen
+import com.ddoczi.tasky.authentication.presentation.registration.RegistrationViewModel
+import com.ddoczi.tasky.core.navigation.Route
 import com.ddoczi.tasky.ui.theme.TaskyTheme
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class MainActivity : ComponentActivity() {
-    private val loginViewModel by viewModels<LoginViewModel>()
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
             TaskyTheme {
-                LoginScreen(
-                    loginViewModel.state.collectAsState().value,
-                    loginViewModel::onEvent
-                )
+                val navController = rememberNavController()
+                TaskyMainScreen(navController, Route.LOGIN)
             }
         }
     }
 }
 
+@Composable
+fun TaskyMainScreen(
+    navController: NavHostController,
+    startDestination: String
+) {
+    NavHost(
+        navController = navController,
+        startDestination = startDestination
+    ) {
+        composable(Route.LOGIN) {
+            val viewModel = hiltViewModel<LoginViewModel>()
+            val state by viewModel.state.collectAsStateWithLifecycle()
+            LoginScreen(
+                state = state,
+                onEvent = viewModel::onEvent,
+                onSignUpClick = { navController.navigate(Route.REGISTRATION) }
```

Comment:
Isn't there already an `OnSignUpClick` event part of the `LoginEvent`? Then you could just catch it here without needing an extra lambda:
```
LoginScreen(
    state = state,
    onEvent = { event ->
        when(event) {
            is LoginEvent.OnSignUpClick -> // Navigate
        }
        viewModel.onEvent(event)
    }
)
```